src/libmongoc/src/mongoc/mongoc-apm.c

```diff
@@ -955,7 +955,9 @@ mongoc_apm_is_sensitive_command (const char *command_name,
       return false;
    }
 
-   return bson_has_field (command, "speculativeAuthenticate");
+   /* An empty command means it has been redacted */
+   return bson_count_keys (command) == 0 ||
+          bson_has_field (command, "speculativeAuthenticate");
```

Comment:
`mongoc_apm_is_sensitive_command` is called in two places:

1. `mongoc_apm_command_started_init` to determine if the command should be redacted.
2. `is_sensitive_command` in entity-map.c, called by `should_ignore_event` to determine if the command should be ignored instead of captured.

(1) should not be calling this with an empty document.

Should we add the empty document check in `is_sensitive_command`?

Reply:
Probably, yes.

Reply:
I went a different route to clean things up a bit.